### ISSUE TITLE
[Gitalk:/hardware/en/maixzero/m0s/m0s.html] M0S module - Sipeed Wiki

### ISSUE THREAD

**Issue by viewpointsa** [sipeed/sipeed_wiki#404]
https://wiki.sipeed.com/hardware/en/maixzero/m0s/m0s.html

**Comment by viewpointsa** [comment]
How can I start with my M0s dock board ?
I plug-it but no communication with serial. I try maixpy IDE, nothing work...
How to start ? How to flash micropython firmware, How to use bluetooth ?

**Comment by wonderfullook** [comment]
> @viewpointsa
> How can I start with my M0s dock board ?
> I plug-it but no communication with serial. I try maixpy IDE, nothing work...
> How to start ? How to flash micropython firmware, How to use bluetooth ?
> 

This chip is not K210, so it's not supported with MaixPy.
It can be programmer and compiled with [bl_mcu_sdk](https://github.com/bouffalolab/bl_mcu_sdk).
It's not supported micropython but [pikapython](http://pikascript.com/).
Wireless SDK will release in the future, not now.

**Comment by hachat** [comment]
There is 
> @viewpointsa
> How can I start with my M0s dock board ?
> I plug-it but no communication with serial. I try maixpy IDE, nothing work...
> How to start ? How to flash micropython firmware, How to use bluetooth ?
> 


> @wonderfullook
> > @viewpointsa
> > How can I start with my M0s dock board ?
> > I plug-it but no communication with serial. I try maixpy IDE, nothing work...
> > How to start ? How to flash micropython firmware, How to use bluetooth ?
> > 
> 
> This chip is not K210, so it's not supported with MaixPy.
> It can be programmer and compiled with [bl_mcu_sdk](https://github.com/bouffalolab/bl_mcu_sdk).
> It's not supported micropython but [pikapython](http://pikascript.com/).
> Wireless SDK will release in the future, not now.

I managed to get it on and the blinker program working. Yet, there is no information about the Wifi.
When will be the Wireless SDK will be available? I wanted to try Wifi6 with M0s!

**Comment by wonderfullook** [comment]
> @viewpointsa
> Ok I find two usefull documentation : 
> 
> - https://bl-mcu-sdk.readthedocs.io/zh_CN/latest/get_started/flash_prog.html#
> - https://github.com/bouffalolab/bl_mcu_sdk
> 
> I have download bl_mcu_sdk & toolchain_gcc_t-head_windows, to build helloworld exemple, but now I cannot flash my board with this command:
> ```
> bl_mcu_sdk\examples\helloworld>make flash CHIP=bl616 COMX=COM8 BAUDRATE=20000
> [17:30:56.270] - {"ErrorCode": "0050","ErrorMsg":"BFLB IMG LOAD SHAKEHAND FAIL"}
> ```
> Same result by Bouffalo Lab Dev Cube 1.8.3
> 

Press boot key and power up M0s Dock to make it into download mode

**Comment by viewpointsa** [comment]
Ok to go in boot mode you must unplug and plug with boot button push. Unplug it after flash to reset
I have flash bl_mcu_sdk\examples\peripherals\gpio\gpio_input_output with modification on IO27, and that work.
How I can dialog with board, I try with Rx Tx (J4 connetor), no data. How to read from USB or Uart ?

**Comment by WitoTV** [comment]
Is there any time esimation for the release of Wireless SDK? I have a little personal project I would really like to do... I'm even willing to sign an NDA if it would allow me to play with it...

**Comment by viewpointsa** [comment]
Yes the use of a Bluetooth and Wi-Fi chip, which can't do wireless, is not very useful.

**Comment by bshambaugh** [comment]
Perhaps this is wifi6? https://github.com/bouffalolab/bouffalo_sdk/blob/79bedc2b8dfde69ac64cf20c012b27848f8f03be/components/wireless/wifi6/inc/wifi_mgmr_ext.h

**Comment by aamiel1** [comment]
Hi,

How can I use this device with as a Zegbee sensor?

**Comment by ShojiMiyanishi** [comment]
I want to make case for M0S dock.
Where is 3D model file of M0S dock?
I found M0s model but M0s dock are not there.

**Comment by Alejandroperezm** [comment]
> @viewpointsa
> Ok to go in boot mode you must unplug and plug with boot button push. Unplug it after flash to reset
> I have flash bl_mcu_sdk\examples\peripherals\gpio\gpio_input_output with modification on IO27, and that work.
> How I can dialog with board, I try with Rx Tx (J4 connetor), no data. How to read from USB or Uart ?

I have the same problem, any clue about this?

**Comment by matthiasnowak** [comment]
Is there ANY ETA on the Wifi, Bluetooth and/or zigbee libraries?
It really is shady and ungrateful for your customers to advertise the m0s as having zigbee/wifi/bluetooth for OVER a year without having actual support to use it!

**Comment by jimoquinn** [comment]
Where is the Zigbee support?  It is the only reason I purchased this board.  Maybe I'm looking in the wrong directory in the repository, so can you point me to the Zigbee support?

**Comment by JanithaB** [comment]
> Ok to go in boot mode you must unplug and plug with boot button push. Unplug it after flash to reset I have flash bl_mcu_sdk\examples\peripherals\gpio\gpio_input_output with modification on IO27, and that work. How I can dialog with board, I try with Rx Tx (J4 connetor), no data. How to read from USB or Uart ?

I also have this problem. I tried to flash the "hello world" example, and it was uploaded successfully. But after "reset", as you said by unplugging, it didn't show up in any "COMPORT" to view in the serial port tool.

**Comment by TronicsMate** [comment]
what are the BT profiles this SDK support?
